src/Core/Cache/DefaultCacheFactory.php

```diff
@@ -114,20 +117,52 @@ protected function isPHPFilesSupported()
     }
 
     /**
-     * @param string $class
-     * @param array $args
-     * @return CacheInterface
+     * Creates an object with a PSR-16 interface, usually from a PSR-6 class name
+     *
+     * Quick explanation of caching standards:
+     * - Symfony cache implements the PSR-6 standard
+     * - Symfony provides adapters wrap a PSR-6 backend with a PSR-16 interface
+     * - Silverstripe uses the PSR-16 to interface with caches
+     * - Psr\SimpleCache\CacheInterface is the php interface of the PSR-16 standard
+     *
+     * Further reading:
+     * - https://symfony.com/doc/current/components/cache/psr6_psr16_adapters.html#using-a-psr-6-cache-object-as-a-psr-16-cache
+     * - https://github.com/php-fig/simple-cache
      */
-    protected function createCache($class, $args)
+    protected function createCache(string $class, array $args, bool $useInjector = true): CacheInterface
     {
-        /** @var CacheInterface $cache */
-        $cache = Injector::inst()->createWithArgs($class, $args);
-
-        // Assign cache logger
-        if ($this->logger && $cache instanceof LoggerAwareInterface) {
-            $cache->setLogger($this->logger);
+        $classIsPsr6 = is_a($class, CacheItemPoolInterface::class, true);
+        $classIsPsr16 = is_a($class, CacheInterface::class, true);
+        if (!$classIsPsr6 && !$classIsPsr16) {
+            throw new InvalidArgumentException("class $class must be PSR-6 or PSR-16 compatible");
         }
-
-        return $cache;
+        if ($classIsPsr6) {
+            if ($useInjector) {
+                // Injector is used for in most instances to allow modification of the cache implementations
+                $psr6Cache = Injector::inst()->createWithArgs($class, $args);
+            } else {
+                // ManifestCacheFactory cannot use Injector because config is not available at that point
+                $psr6Cache = new $class(...$args);
+            }
+
+            // Assign cache logger
+            if ($this->logger && $psr6Cache instanceof LoggerAwareInterface) {
+                $psr6Cache->setLogger($this->logger);
+            }
```

Comment:
Why are we only doing this for psr6 cache?

Reply:
Have added logic to also try adding it to the Psr16Cache, if it hasn't already been added to an inner Psr6Cache